lib/embeds/services/image.utils.dart

```diff
@@ -50,18 +51,61 @@ Image imageByUrl(String imageUrl,
   }
 
   if (imageUrl.startsWith('http')) {
-    return Image.network(imageUrl,
-        width: width, height: height, alignment: alignment);
+    return Image.network(
+      imageUrl,
+      width: width,
+      height: height,
+      alignment: alignment,
+      loadingBuilder: (context, child, loadingProgress) {
+        if (loadingProgress == null) return child;
+        return Center(
+          child: CircularProgressIndicator(
+            value: loadingProgress.expectedTotalBytes != null
+                ? loadingProgress.cumulativeBytesLoaded /
+                    loadingProgress.expectedTotalBytes!
+                : null,
+          ),
+        );
+      },
+      errorBuilder: (context, exception, stackTrace) {
+        return Row(
+          children: [
+            Icon(Icons.error, color: Colors.redAccent),
+            SizedBox(width: 16),
+            Text(exception.toString()),
+          ],
+        );
+      },
+    );
   }
   return Image.file(io.File(imageUrl),
       width: width, height: height, alignment: alignment);
 }
 
-String standardizeImageUrl(String url) {
+String standardizeMediaUrl(String url) {
   if (url.contains('base64')) {
     return url.split(',')[1];
   }
-  return url;
+
+  // Process image URL to embed.
+
+  // Processes Google Drive image to correct format.
+  if (url.contains('https://drive.google.com/file/d/')) {
+    url = url.replaceAll(
+        'https://drive.google.com/file/d/', 'https://drive.google.com/uc?id=');
+  }
+
+  // Changes Discord domain
+  if (url.contains('cdn.discordapp.com')) {
+    url = url.replaceAll('cdn.discordapp.com', 'media.discordapp.net');
+  }
+
+  // Uses CORS proxy for web
```

Comment:
I don't believe this should be part of the library. There should be a configuration feature. And a nice thought trough, API for it.